node/core/approval-voting/src/criteria.rs

```diff
@@ -97,6 +100,53 @@ fn relay_vrf_modulo_transcript(relay_vrf_story: RelayVRFStory, sample: u32) -> T
 	t
 }
 
+/// A hard upper bound on num_cores * target_checkers / num_validators
+const MAX_MODULO_SAMPLES: usize = 40;
+
+use std::convert::AsMut;
+
+fn clone_into_array<A, T>(slice: &[T]) -> A
+where
+	A: Default + AsMut<[T]>,
+	T: Clone,
+{
+	let mut a = A::default();
+	<A as AsMut<[T]>>::as_mut(&mut a).clone_from_slice(slice);
+	a
+}
+
+struct BigArray(pub [u8; MAX_MODULO_SAMPLES * 4]);
+
+impl Default for BigArray {
+	fn default() -> Self {
+		BigArray([0u8; MAX_MODULO_SAMPLES * 4])
+	}
+}
+
+impl AsMut<[u8]> for BigArray {
+	fn as_mut(&mut self) -> &mut [u8] {
+		self.0.as_mut()
+	}
+}
+
+/// Return an iterator to all core indices we are assigned to.
+fn relay_vrf_modulo_cores(
+	vrf_in_out: &VRFInOut,
+	// Configuration - `relay_vrf_modulo_samples`.
+	num_samples: u32,
+	// Configuration - `n_cores`.
+	max_cores: u32,
+) -> Vec<CoreIndex> {
+	vrf_in_out
+		.make_bytes::<BigArray>(approval_types::CORE_RANDOMNESS_CONTEXT)
+		.0
+		.chunks_exact(4)
+		.take(num_samples as usize)
+		.map(move |sample| CoreIndex(u32::from_le_bytes(clone_into_array(&sample)) % max_cores))
+		.unique()
+		.collect::<Vec<CoreIndex>>()
```

Comment:
How big is big array?  You `make_merlin_rng` is kinda ugly code, but `make_chacharng` is fairly clean code: https://docs.rs/schnorrkel/latest/src/schnorrkel/vrf.rs.html#373  You can then extract sequentially. 

Also an `unsafe` lifetime works.  lol
```
let mut ctx = [0u8; LEN + 4];
ctx[0..LEN].copy_from_slice( approval_types::CORE_RANDOMNESS_CONTEXT );
for i in foo {
        ctx[LEN..].copy_from_slice( & i.to_le_bytes() );
	vrf_in_out
		.make_bytes::<BigArray>(unsafe { &ctx as *[u8] as &'static [u8] })
                ...
}
```

Reply:
oh I see it's just 40 samples, or 160 bytes.  Yeah just do what you have here.  Ignore my comment above.